### PR TITLE
Accelerate Accuracy-aware Tuning by Leveraging Multi-nodes 

### PR DESCRIPTION
**Pull request by yiliu30**
# What does this PR do?

Introduce the [Intel Neural Compressor distributed tuning](https://github.com/intel/neural-compressor/blob/master/docs/source/tuning_strategies.md#distributed-tuning) and enable it with one example.

One challenge in model quantization is identifying the optimal configuration that balances accuracy and performance. Distributed tuning speeds up this time-consuming process by parallelizing it across multiple nodes.

Below is a diagram that illustrates how it works.

distributed_tuning_intro

## Before submitting
- [x] Introduce the Intel Neural Compressor distributed tuning
- [x] Enable it with [one example](https://github.com/yiliu30/optimum-intel/tree/ly/dist_inc/examples/neural_compressor/text-classification#distributed-quantization-with-multi-node)
- [x] Updated the documentation accordingly
- [x] The test has been covered by Intel Neural Compressor

### CODE REVIEW

**Comment by yiliu30** [comment]
Hi @hshen14 @ftian1 @PenghuiCheng , the PR is ready, please review it. May I know who else I should mention to review it?

**Comment by ftian1** [comment]
@echarlaix pls check if this is ok for you. we would like to add this feature by leveraging multi-nodes to boost up the accuracy aware tuning process.

File: examples/neural_compressor/text-classification/README.md [hunk 1]
```diff
@@ -42,6 +44,41 @@ python run_glue_post_training.py \
...
+> **_Note:_** User can also install Open MPI with [Conda](https://anaconda.org/conda-forge/openmpi).
+
+In `run_glue_post_training.py`, set `quant_level` to 1 as the following statement:
+
+```python
+from neural_compressor import PostTrainingQuantConfig
+quantization_config = PostTrainingQuantConfig(approach=optim_args.quantization_approach, quant_level=1)
+```
+
+> **_Note:_** Please also add below code snippet at the beginning of the example to disable the initialization of torch distribution.
```
Comment (echarlaix) [inline examples/neural_compressor/text-classification/README.md:65]:
```suggestion
> **_Note:_** Please also add the below code snippet at the beginning of the example to disable the initialization of torch distribution.
```

File: examples/neural_compressor/text-classification/README.md [hunk 2]
```diff
@@ -42,6 +44,41 @@ python run_glue_post_training.py \
...
+
+And then, modify the run_task_in_distributed_mode.sh according to the cluster information. Below is the explanation for each parameter.
+
+- `<NUM_PROCESS>` is the number of processes, recommend to set it with the number of hosts plus one.
+
+- `<MAX_NUM_THREADS>` is the number of threads, recommend to set it with the number of physical cores on one node.
+
+- `<HOSTNAME>` is the host name, and argument --host `<HOSTNAME>`,`<HOSTNAME>`... can be replaced with `--hostfile <HOSTFILE>`, when each line in `<HOSTFILE>` is a host name.
+
+- `-mca btl_tcp_if_include <NETWORK_INTERFACE>` is used to set the network communication interface between hosts. For example, `<NETWORK_INTERFACE>` can be set to `192.168.20.0/24` to allow MPI communication between all hosts under `192.168.20.*`network segment.
```
Comment (echarlaix) [inline examples/neural_compressor/text-classification/README.md:77]:
```suggestion
- `-mca btl_tcp_if_include <NETWORK_INTERFACE>` is used to set the network communication interface between hosts. For example, `<NETWORK_INTERFACE>` can be set to `192.168.20.0/24` to allow MPI communication between all hosts under `192.168.20.*` network segment.
```

**Review by echarlaix — APPROVED** [review]
LGTM, left a couple of minor comments, thanks for the addition !

**Comment by echarlaix** [comment]
Thanks @yiliu30 ! Also before we merge, can you run the `make style` command to to fix the code quality issues ? To install the required dependencies you can run `pip install .[quality]`

**Comment by yiliu30** [comment]
> Thanks @yiliu30 ! Also before we merge, can you run the `make style` command to to fix the code quality issues ? To install the required dependencies you can run `pip install .[quality]`

@echarlaix Thanks for your guidance. I fixed it with [a9fa8a6](https://github.com/huggingface/optimum-intel/pull/360/commits/a9fa8a6bbcf1552e7eb07d73ef0848f19b207e04).
Here are my local test results:
```shell
$ make style
black .
Skipping .ipynb files as Jupyter dependencies are not installed.
You can fix this by running ``pip install "black[jupyter]"``
All done! ✨ 🍰 ✨
117 files left unchanged.
ruff . --fix
```

**Comment by yiliu30** [comment]
Hi @echarlaix, could you please retrigger the test? thx :)

**Comment by yiliu30** [comment]
Hi @echarlaix, some OpenVINO-related checks have failed. Should I rebase the master branch to fix it?

**Comment by echarlaix** [comment]
> Hi @echarlaix, some OpenVINO-related checks have failed. Should I rebase the master branch to fix it?

yes

**Pull request merged**